test/RibbonThetaVault.ts

```diff
@@ -3322,6 +3322,122 @@ function behavesLikeRibbonOptionsVault(params: {
       });
     });
 
+    describe("#stake", () => {
+      let stakingRewards: Contract;
+
+      time.revertToSnapshotAfterEach(async () => {
+        const MockStakingRewards = await getContractFactory(
+          "MockStakingRewards",
+          ownerSigner
+        );
+        stakingRewards = await MockStakingRewards.deploy(vault.address);
+      });
+
+      it("reverts when stakingRewards is not set", async function () {
+        await assetContract
+          .connect(userSigner)
+          .approve(vault.address, depositAmount);
+        await vault.deposit(depositAmount);
+        await rollToNextOption();
+        await expect(vault.stake(depositAmount)).to.be.revertedWith(
+          "!stakingRewards"
+        );
+      });
+
```

Comment:
you should add a test case for 1. deposit 2. maxRedeem 3. stake(prev redeemed shares)

Reply:
Added: https://github.com/ribbon-finance/ribbon-v2/pull/194/commits/92bfe9b03ee5c687cb20a46e179a713040d6aee6